lib/network_tools.cpp

```diff
@@ -127,35 +127,40 @@ using Config = vector<Wiring>;
 
 using Configs = vector<Config>;
 
-Matrix addSplitter(Matrix network, const Wiring splitter_inputs,
-                   const Wiring splitter_outputs) {
-  const int N = network.size();
-  const int M = network[0].size();
+Matrix addSplitterToFlow(Matrix flow, const Wiring splitter_inputs,
+                         const Wiring splitter_outputs) {
+  const int num_flow_outputs = flow.size();
+  const int num_flow_inputs = flow[0].size();
 
-  const int n = splitter_inputs.size();
-  const int m = splitter_outputs.size();
+  const int num_splitter_outputs = splitter_outputs.size();
+  const int num_splitter_inputs = splitter_inputs.size();
 
-  Row splitter_flow = zeroRow(M);
-  for (int i = 0; i < n; ++i) {
-    splitter_flow.push_back(1.0 / m);
+  Row splitter_flow = zeroRow(num_flow_inputs);
+  for (int i = 0; i < num_splitter_inputs; ++i) {
+    splitter_flow.push_back(1.0 / num_splitter_outputs);
   }
 
   // Find this splitter's output in terms of its input flows
-  for (int i = 0; i < n; ++i) {
+  for (int i = 0; i < num_splitter_inputs; ++i) {
     if (splitter_inputs[i] != -1) {
-      for (int j = 0; j < M; ++j) {
-        splitter_flow[j] += network[splitter_inputs[i]][j] / m;
+      for (int j = 0; j < num_flow_inputs; ++j) {
+        splitter_flow[j] += flow[splitter_inputs[i]][j] / num_splitter_outputs;
       }
     }
   }
 
   // Remove circular dependencies of the new outputs on any inputs that they
   // lead to
   Row new_splitter_flow = splitter_flow;
-  for (int i = 0; i < m; ++i) {
+  for (int i = 0; i < num_splitter_outputs; ++i) {
     if (splitter_outputs[i] != -1) {
-      for (int j = 0; j < n + M; ++j) {
+      for (int j = 0; j < num_flow_inputs + num_splitter_inputs; ++j) {
         new_splitter_flow[j] *= 1 / (1 - splitter_flow[splitter_outputs[i]]);
+
+        if (splitter_flow[splitter_outputs[i]] == 1) {
+          assert(1 == 0);  // Verify this branch never runs
```

Comment:
Good idea adding this assert. Kept this in by accident for debugging, but checking that we don't have NaN's creep in would be a good thing to do.